Firebase/ios/Bonfire/Core/Channel/PersistedChannelsService.swift

```diff
@@ -0,0 +1,104 @@
+import Foundation
+import RxSwift
+
+class PersistedChannelsService: ChannelsService {
```

Comment:
These `PersistedServices` match the Java implementations - they push the business of actually writing to the database out into a `Database` protocol, which allows us to test that we're writing to all the relevant parts of the database (a downside of a synced database like Firebase is the requirement for intimate knowledge of how things are stored on the part of the client, but we can at least make sure it's well defined and tested).